app/containers/GamePage/index.js

```diff
@@ -21,20 +23,11 @@ export default class GamePage extends Component {
     };
   }
 
-  componentWillMount() {
-    const { auth } = this.state;
-    // Initiates Firebase auth and listen to auth state changes.
-    auth.onAuthStateChanged(this.onAuthStateChanged);
-    this.loadMessages();
-  }
-
-  // Triggers when the auth state change for instance when the user signs-in or signs-out.
-  onAuthStateChanged = (user) => {
-    if (user) {
-      this.setState({
-        loggedIn: true,
-        username: user.displayName,
-      });
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.loggedIn) {
+      this.loadMessages();
+    } else {
```

Comment:
I'm pretty sure this should be nextProps.loggedOut because this componentWillReceiveProps can receive other types of props too